oan-rest/src/test/java/org/jax/oan/utils/QueryCleanerTest.java

```diff
@@ -0,0 +1,16 @@
+package org.jax.oan.utils;
+
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+class QueryCleanerTest {
+
+	@Test
+	void clean() {
```

Comment:
Suggest that if the string is not in the form expected, e.g. if the replaceAll results in empty string or the input does not contain a ":" that an IllegalArgumentException is thrown. Then the test checks that invalid strings throw the correct exception.

```
public static String clean(String query) {
               if (query==null) throw new IllegalArgumentException(...);
               if (query.indexOf(":")<0) throw new IllegalArgumentException(...);
               // TODO Could use matcher and check string matches too.
               String res = query.replaceAll("[^a-zA-Z0-9\\-:\\s]", "");
               if (res.isBlank()) throw new IllegalArgumentException(...);
               return res;
}
```